pkg/apiserver/certificate/selfsignedcert_provider.go

```diff
@@ -119,6 +119,9 @@ func (p *selfSignedCertProvider) RunOnce(ctx context.Context) error {
 }
 
 func (p *selfSignedCertProvider) Run(ctx context.Context, workers int) {
+	var wg sync.WaitGroup
+	// ensure goroutines have exited before returning, to simplify unit tests.
```

Comment:
@tnqn if you don't think this change to the Run function is warranted, I'm happy with considering alternatives